modules/javafx.controls/src/main/java/javafx/scene/control/ResizeFeaturesBase.java

```diff
@@ -96,7 +97,24 @@ public ResizeFeaturesBase(@NamedArg("column") TableColumnBase<S,?> column, @Name
    *
    * @since 20
    */
-  public void setColumnWidth(TableColumnBase<S,?> col, double width) {
-      col.doSetWidth(width);
+  public void setColumnWidth(TableColumnBase<S, ?> col, double width) {
+      Control c = getTableControl();
+      if (c.isSnapToPixel()) {
+          double min = c.snapSizeX(col.getMinWidth());
+          double max = RegionHelper.snapInnerSpaceX(c, col.getMaxWidth());
```

Comment:
I'm still not convinced that min and max should be snapped differently (floor vs ceiling).

Reply:
we will violate the max constraint if we snap to a larger value.

so for the max constraint, we should `snapInnerSpace()` which is `Region.snapPortionX()`.

Reply:
> we will violate the max constraint if we snap to a larger value.

Well, it's inherently an over-constrained problem. Conceptually, though, this does not violate the constraint. When snap-to-pixel is enabled, all values, including min and max values are snapped to a pixel that can be almost 1 pixel larger or smaller than it's given value. Within that tolerance, we are honoring the min  and max values.

It isn't required that we always ceil the min value and floor the max value (not to mention that doing that is what causes the odd min > max).

So I don't buy this argument. Is there some other reason to do it?

Reply:
no other reason.  are you suggesting to simply `round()`?